### PR TITLE
STIG Update RHEL/7: Add new SSHD and AIDE XCCDF content

### PR DESCRIPTION
**Pull request by redhatrises**
- Add XCCDF for RHEL-07-020140, RHEL-07-021600, RHEL-07-021610, RHEL-07-021620,
  RHEL-07-040332, and RHEL-07-040333
- Fix STIG Rule mapping for RHEL-07-020130
- Add new XCCDF rules to STIG profile

### CODE REVIEW

**Comment by redhatrises** [comment]
@mpreisler I believe that the el6 Jenkins node is offline.

**Comment by mpreisler** [comment]
@redhatrises should be fixed now

**Comment by redhatrises** [comment]
[retest this please]

**Comment by redhatrises** [comment]
[retest this please]

**Comment by redhatrises** [comment]
Fixes #1233

File: shared/xccdf/services/ssh.xml [hunk 1]
```diff
@@ -465,6 +465,53 @@ can allow an attacker to move trivially to other hosts.
...
+</description>
+<ocil>
+<sshd-check-macro option="IgnoreUserKnownHosts" value="yes" default="no" />
+</ocil>
+<rationale>
+Configuring this setting for the SSH daemon provides additional
+assurance that remove login via SSH will require a password, even
+in the event of misconfiguration elsewhere.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
```
Comment (shawndwells) [inline shared/xccdf/services/ssh.xml]: Mapped this to CCE-80372-6 via commit 3c48078

File: shared/xccdf/services/ssh.xml [hunk 2]
```diff
@@ -465,6 +465,53 @@ can allow an attacker to move trivially to other hosts.
...
+</description>
+<ocil>
+<sshd-check-macro option="RhostsRSAAuthentication" value="no" default="yes" />
+</ocil>
+<rationale>
+Configuring this setting for the SSH daemon provides additional
+assurance that remove login via SSH will require a password, even
+in the event of misconfiguration elsewhere.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
```
Comment (shawndwells) [inline shared/xccdf/services/ssh.xml]: Mapped CCE-80373-4 to sshd_disable_rhosts_rsa via ea935c9

File: shared/xccdf/system/software/integrity.xml [hunk 3]
```diff
@@ -98,23 +98,153 @@ must be captured and it should be able to be verified against the installed file
...
+operating system configurations can have unintended side effects, some of which may
+be relevant to security.
+<br/><br/>
+Detecting such changes and providing an automated response can help avoid unintended,
+negative consequences that could ultimately affect the security state of the operating
+system. The operating system's Information Management Officer (IMO)/Information System
+Security Officer (ISSO) and System Administrators (SAs) must be notified via email and/or
+monitoring system trap when there is an unauthorized modification of a configuration item.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
```
Comment (shawndwells) [inline shared/xccdf/system/software/integrity.xml]: Mapped CCE-80374-2 to aide_scan_notification via 475428d

File: shared/xccdf/system/software/integrity.xml [hunk 4]
```diff
@@ -98,23 +98,153 @@ must be captured and it should be able to be verified against the installed file
...
+<ocil clause="the acl option is missing or not added to the correct ruleset">
+To determine that AIDE is verifying ACLs, run the following command:
+<pre>$ grep acl /etc/aide.conf</pre>
+Verify that the <tt>acl</tt> option is added to the correct ruleset.
+</ocil>
+<rationale>
+ACLs can provide permissions beyond those permitted through the file mode and must be
+verified by the file integrity tools.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
```
Comment (shawndwells) [inline shared/xccdf/system/software/integrity.xml]: Mapped CCE-80375-9 to aide_verify_acls to bacee36

File: shared/xccdf/system/software/integrity.xml [hunk 5]
```diff
@@ -98,23 +98,153 @@ must be captured and it should be able to be verified against the installed file
...
+To determine that AIDE is verifying extended file attributes, run the following command:
+<pre>$ grep xattrs /etc/aide.conf</pre>
+Verify that the <tt>xattrs</tt> option is added to the correct ruleset.
+</ocil>
+<rationale>
+Extended attributes in file systems are used to contain arbitrary data and file metadata
+with security implications.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
+<oval id="aide_verify_ext_attributes"/>
```
Comment (shawndwells) [inline shared/xccdf/system/software/integrity.xml]: Mapped CCE-80376-7 to aide_verify_ext_attributes via 4829638

File: shared/xccdf/system/software/integrity.xml [hunk 6]
```diff
@@ -98,23 +98,153 @@ must be captured and it should be able to be verified against the installed file
...
+To determine that AIDE is configured for FIPS 140-2 file hashing, run the following command:
+<pre>$ grep sha512 /etc/aide.conf</pre>
+Verify that the <tt>sha512</tt> option is added to the correct ruleset.
+</ocil>
+<rationale>
+File integrity tools use cryptographic hashes for verifying file contents and directories
+have not been altered. These hashes must be FIPS 140-2 approved cryptographic hashes.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
+<oval id="aide_use_fips_hashes"/>
```
Comment (shawndwells) [inline shared/xccdf/system/software/integrity.xml]: Opened https://github.com/OpenSCAP/scap-security-guide/issues/1563 to track this needs OVAL

File: shared/xccdf/system/software/integrity.xml (hunk 5 above)
Comment (shawndwells) [inline shared/xccdf/system/software/integrity.xml]: Opened https://github.com/OpenSCAP/scap-security-guide/issues/1564 to track this needs OVAL

File: shared/xccdf/system/software/integrity.xml [hunk 7]
```diff
@@ -98,23 +98,153 @@ must be captured and it should be able to be verified against the installed file
...
+To determine that AIDE is verifying ACLs, run the following command:
+<pre>$ grep acl /etc/aide.conf</pre>
+Verify that the <tt>acl</tt> option is added to the correct ruleset.
+</ocil>
+<rationale>
+ACLs can provide permissions beyond those permitted through the file mode and must be
+verified by the file integrity tools.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
+<oval id="aide_verify_acls"/>
```
Comment (shawndwells) [inline shared/xccdf/system/software/integrity.xml]: Opened https://github.com/OpenSCAP/scap-security-guide/issues/1565 to track OVAL generation

File: shared/xccdf/system/software/integrity.xml [hunk 8]
```diff
@@ -98,23 +98,153 @@ must be captured and it should be able to be verified against the installed file
...
+be relevant to security.
+<br/><br/>
+Detecting such changes and providing an automated response can help avoid unintended,
+negative consequences that could ultimately affect the security state of the operating
+system. The operating system's Information Management Officer (IMO)/Information System
+Security Officer (ISSO) and System Administrators (SAs) must be notified via email and/or
+monitoring system trap when there is an unauthorized modification of a configuration item.
+</rationale>
+<ident prodtype="rhel7" cce="CCE-TBD" />
+<oval id="aide_scan_notification"/>
```
Comment (shawndwells) [inline shared/xccdf/system/software/integrity.xml]: Opened https://github.com/OpenSCAP/scap-security-guide/issues/1566 to track OVAL

**Comment by shawndwells** [comment]
LGTM. Added CCEs to the XCCDF. Opened tickets for rules that need OVAL.

**Pull request merged**